sdk-extension/opentelemetry-sdk-extension-aws/src/opentelemetry/sdk/extension/aws/resource/_lambda.py

```diff
@@ -0,0 +1,66 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from os import environ
+
+from opentelemetry.sdk.resources import Resource, ResourceDetector
+from opentelemetry.semconv.resource import (
+    CloudPlatformValues,
+    CloudProviderValues,
+    ResourceAttributes,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class AwsLambdaResourceDetector(ResourceDetector):
+    """Detects attribute values only available when the app is running on AWS
+    Lambda and returns them in a Resource.
+
+    Uses Lambda defined runtime enivronment variables. See more: https://docs.aws.amazon.com/lambda/latest/dg/configuration-envvars.html#configuration-envvars-runtime
+    """
+
+    def detect(self) -> "Resource":
+        try:
+            # NOTE: (NathanielRN) Should ResourceDetectors use Resource.create() to pull in the environment variable?
+            # `OTELResourceDetector` doesn't do this...
+            return Resource(
+                {
+                    ResourceAttributes.CLOUD_PROVIDER: CloudProviderValues.AWS.value,
+                    ResourceAttributes.CLOUD_PLATFORM: CloudPlatformValues.AWS_LAMBDA.value,
+                    ResourceAttributes.CLOUD_REGION: environ["AWS_REGION"],
```

Comment:
Shouldn't be set if `None` right?

Reply:
Sorry what do you mean here? I added in the docstring for `AwsLambdaResourceDetector` a link to the [Lambda Docs](https://docs.aws.amazon.com/lambda/latest/dg/configuration-envvars.html#configuration-envvars-runtime) which say certain environment variables (e.g. `AWS_REGION`) **must** be set.

I'm counting on this to fail if they're not on a Lambda environment at which point logs could help them figure out what went wrong. (They can ignore it using `self.raise_on_error`).